processor/k8sattributesprocessor/README.md

```diff
@@ -102,6 +102,7 @@ Additional container level attributes can be extracted provided that certain res
    - container.image.tag
    - container.image.repo_digests (if k8s CRI populates [repository digest field](https://github.com/open-telemetry/semantic-conventions/blob/v1.26.0/model/registry/container.yaml#L60-L71))
 2. If the `k8s.container.name` resource attribute is provided, the following additional attributes will be available:
+   - container.id
```

Comment:
This is only kinda true. The next section tries to clarify:

> If the `k8s.container.restart_count` resource attribute is provided, it can be used to associate with a particular container
   instance. If it's not set, the latest container instance will be used
   
You need either `container.id` or `k8s.container.name` provided to get to the [logic](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/145145c70c1a87c4b2bca7a6159686c001414ccc/processor/k8sattributesprocessor/processor.go#L194) that will attempt to use `k8s.pod.restart_count` to associate to the container id and fallback to latest container.

So while it is likely that supplying `k8s.container.name` will get you a `container.id`, it is a guarantee to get the correct `container.id`

Reply:
If we add this here, I'd like some sort of note that points you to the caveat in section 3.

Reply:
@TylerHelmuth, I understand  that containerId may be incorrect if it is data from non-last restarted container withoug `restart_count` info. Do you have in mind any way that can be recommended to users to configure application that produces OTEL data to get `restart_count` data?
It would be relatively simple to get `k8s.container.name`, as it requires static. I don't know a way to resolve `restart_count` easy.

Reply:
Right now the receiver that can reliably set `k8s.container.restart_count` is the filelogreceiver, since it gets the restart count from the file name.

I am ok with including `container.id` in this list, I just want to make sure users who read that line also comprehend the caveat.

Reply:
And you add a `*` or `†` after `container.id` and add the same symbol somewhere in the section below?

Reply:
If I understand correctly, without `k8s.container.restart_count`, it is not guarantee to get the correct `container.id`, right?

Reply:
That is correct.